partiql-types/src/main/kotlin/org/partiql/value/impl/Int16ValueImpl.kt

```diff
@@ -32,6 +49,27 @@ internal data class Int16ValueImpl(
     override fun withAnnotations(annotations: Annotations): Int16Value = _withAnnotations(annotations)
 
     override fun withoutAnnotations(): Int16Value = _withoutAnnotations()
+    override fun toInt8(): Int8Value {
+        val byte = this.value?.toByte() ?: return int8Value(null, annotations)
+        if (byte.toShort() != this.value) {
+            throw DataException("Overflow when casting ${this.value} to INT8")
+        }
+        return int8Value(byte, annotations)
+    }
+
+    override fun toInt16(): Int16Value = this
+
+    override fun toInt32(): Int32Value = int32Value(this.value?.toInt(), annotations)
+
+    override fun toInt64(): Int64Value = int64Value(this.value?.toLong(), annotations)
+
+    override fun toInt(): IntValue = intValue(this.value?.toLong()?.let { BigInteger.valueOf(it) }, annotations)
+
+    override fun toDecimal(): DecimalValue = decimalValue(this.value?.toBigDecimal(), annotations)
+
+    override fun toFloat32(): Float32Value = float32Value(this.value?.toFloat(), annotations)
+
+    override fun toFloat64(): Float64Value = float64Value(this.value?.toDouble(), annotations)
```

Comment:
Random note, and I don't think we need to change anything here. Just wanted to call out that we are preserving annotations on the cast.